src/overlays/actors/ovl_En_Dt/z_en_dt.c

```diff
@@ -28,59 +95,714 @@ ActorInit En_Dt_InitVars = {
+                                      this->targetActor);
+
+                this->timer = 2;
+                this->actionFunc = EnDt_FinishMeetingCutscene;
+                return;
+            }
+
+            if (this->textIdIndex == 12) { // Mayor - "Ah!"
+                if (CHECK_WEEKEVENTREG(WEEKEVENTREG_60_08)) {
+                    EnDt_UpdateVisualState(this);
+                    this->timer = 25;
+                } else {
+                    EnDt_ChangeAnim(this, EN_DT_ANIMATION_4);
+                    SkelAnime_Update(&this->skelAnime);
+                }
+            }
+
+            if (this->textIdIndex == 13) { // Muto - "Ah!"
+                EnDt_ChangeAnim(this, EN_DT_ANIMATION_4);
+                SkelAnime_Update(&this->skelAnime);
+            }
+
+            EnDt_UpdateCutsceneFocusTarget(this);
+            Actor_ChangeFocus(&this->actor, play, this->targetActor);
+            Camera_SetTargetActor(Play_GetCamera(play, CutsceneManager_GetCurrentSubCamId(this->csIds[index])),
+                                  this->targetActor);
+
+            this->actor.textId = sTextIds[this->textIdIndex];
+            Message_CloseTextbox(play);
+            Message_StartTextbox(play, this->actor.textId, &this->actor);
+
+            // Sets the cutscene ID index based off the current textId
+            for (index = 0; index < 24; index += 2) {
+                if (play->msgCtx.currentTextId == CsIndexTable.D_80BEB268[index] ||
+                    this->actor.textId == CsIndexTable.D_80BEB268[index]) {
+                    this->csIdIndex = index;
+                    break;
+                }
+            }
+
+            index = CsIndexTable.D_80BEB26A[this->csIdIndex];
+            if (this->cutsceneState == EN_DT_CS_STATE_PLAYING && (index != csIdIndex)) {
+                this->cutsceneState = EN_DT_CS_STATE_WAITING;
+                CutsceneManager_Stop(this->csIds[csIdIndex]);
+                CutsceneManager_Queue(this->csIds[index]);
+            }
+
+            // Mutoh turns his head to focus on the Mayor after saying "Isn't that right Mayor?"
+            if (this->textIdIndex == 3 && this->npcEnMuto != NULL && this->npcEnBaisen != NULL) {
+                muto->targetActor = &this->actor;
+            }
+        }
+    }
+}
+
+void EnDt_FinishMeetingCutscene(EnDt* this, PlayState* play) {
+    f32 currFrame = this->skelAnime.curFrame;
+
+    if (this->timer != 0) {
+        if (this->timer == 1) {
+            if ((this->textIdIndex == 8) || (this->visualState == 4)) {
+                EnDt_UpdateVisualState(this);
+            }
+
+            this->visualState++;
+            if (this->visualState > 5) {
+                this->visualState = 5;
+            }
+        }
+    } else if (this->animEndFrame <= currFrame) {
+        Camera* pCamera;
+        s32 index;
+        s32 currTextId = CsIndexTable.D_80BEB26A[this->csIdIndex];
```

Comment:
```suggestion
        s32 csIdIndex = CsIndexTable.D_80BEB26A[this->csIdIndex];
```
Doesn't seem to be used as a textId, but like a `csIdIndex` in other functions.